### ISSUE TITLE
Ubuntu 18.04 support

### ISSUE THREAD

**Issue by Lirt** [nusenu/ansible-relayor#159]
Hello,

Ubuntu 18.04 will get into [stable release at April 26th](https://wiki.ubuntu.com/BionicBeaver/ReleaseSchedule). I was trying to test it if we can support it. I tried to test it using our test-kitchen by adding platform `ubuntu-18.04` but so far I couldn't make it work.

There are 2 issues I encountered:

1. Ubuntu is now using `netplan` for networking configuration. Scenario with 2 IP addresses (or explicitly set IPs) is not working with vagrant version lower than 2.0.3. There are multiple issues closed in vagrant repository already, eg. https://github.com/hashicorp/vagrant/issues/9134.
2. Tor systemd service fails to run and playbook fails at `Ensure Tor instances are enabled and started (Linux/systemd)`.

To elaborate second point, running tor service fails with these messages.

```sh
Directory /var/lib/tor-instances/10.0.2.15_9000 cannot be read: Permission denied
Can't create/check datadirectory /var/lib/tor-instances/10.0.2.15_9000
Error initializing keys; exiting
```

Directories have permissions set like this:

```sh
root@t-guard-ubuntu-1804:~# ls -la /var/lib/tor-instances/
total 16
drwxr-xr-x  4 root                root                4096 Apr 22 14:01 .
drwxr-xr-x 38 root                root                4096 Apr 22 14:01 ..
drwx------  4 _tor-10.0.2.15_9000 _tor-10.0.2.15_9000 4096 Apr 22 14:01 10.0.2.15_9000
drwx------  4 _tor-10.0.2.15_9100 _tor-10.0.2.15_9100 4096 Apr 22 14:01 10.0.2.15_9100
```

Tor user is existing:

```sh
root@t-guard-ubuntu-1804:~# id _tor-10.0.2.15_9000
uid=998(_tor-10.0.2.15_9000) gid=998(_tor-10.0.2.15_9000) groups=998(_tor-10.0.2.15_9000)
```
Tor configuration has user specified:

```sh
root@t-guard-ubuntu-1804:~# grep "User" /etc/tor/instances/10.0.2.15_9000/torrc
User _tor-10.0.2.15_9000
```

I compared systemd services across Ubuntu 16 and Ubuntu 18 and they are the same (this is obvious since we install same tor package I guess). I though that the hardening part may cause this.

Second thought I have is that this has something to do with apparmor, so I disabled it (`systemctl disable apparmor && reboot) to test it quickly and it didn't help.

Do you have any idea, why can this fail?

I plan to rebuild my few relay and exit nodes from Ubuntu 16 servers to 18 after it is released to stable and vagrant tests will pass. I can report if everything works in production after that.

**Comment by nusenu** [comment]
Hello Ondrej,

thank you for working on this!

The problem is caused by the fact that relayor enables Sandboxing by default on Debian and Ubuntu.

Sandbox appears to have a problem with these new libc (2.27) versions on Ubuntu 18.04.

if you disable Sandboxing (tor_Sandbox: 0) it works just fine (tested: t-guard-ubuntu-1804)

Depending on when the upstream bug is fixed and released we might just
disable Sandboxing by default on Ubuntu 18.04.

 
Upstream bug:
https://trac.torproject.org/projects/tor/ticket/25440

**Comment by Lirt** [comment]
Since you officially do not support Ubuntu 18 yet, I will disable sandboxing in my environment for now as a hotfix and we can leave this issue open so anybody experiencing this on Ubuntu 18 will find information about that.

**Comment by nusenu** [comment]
The ubuntu1804 branch adds support for 18.04 and drops support for 16.04 and will be merged into master once ubuntu 18.04 is released (scheduled for 2018-04-26).

**Comment by nusenu** [comment]
fixed in 9a5beacf7bcfd5b46e3ff0737f1a1dc11d07164f
